### ISSUE TITLE
Show a summary of event associated entities on the event information

### ISSUE THREAD

**Issue by kwa20** [SORMAS-Foundation/SORMAS-Project#5567]
### Situation Description
Thus far, it is hard to get concise information about event participants and associated contacts and cases in the event information itself. To aid the user to have a rough overview of contained entities and potentially help in decision making in regards to disease control activities, implementing an informative section might help.

### Feature Description
Add an information box on the top right of the event information that displays information about entities that are contained in the event. These entities include event participants, associated contacts and resulting cases.

- Add a feature configuration `EVENT_PARTICIPANT_SUMMARY` to activate/deactivate this feature; default = true.

- Add the total number of event participants and break these numbers further down into tested ep's and ep's with vaccination information
    - tested event participants - total number of tested ep's independent of status (percentage of total ep number)
       - positive - total number (percentage of tested ep's)
       - negative - total number (percentage of tested ep's)
       - pending - total number (percentage of tested ep's)
       - indeterminate - total number (percentage of tested ep's)
   - vaccination status - total number of ep's with entered vaccination status (percentage of total ep number)
       - vaccinated - total number (percentage of ep with vaccination status not null)
       - unvaccinated - total number (percentage of ep with vaccination status not null)
       - unknown - total number (percentage of ep with vaccination status not null)

- Add the total number of cases contained in the event and break these numbers further down into case classifications and disease variants
    - Case classification - total number of classified cases
       - suspect - total number of suspected cases (percentage of suspect cases of total case count)
       - probable - total number of probable cases (percentage of probable cases of total case count)
       - confirmed - total number of confirmed cases (percentage of  confirmed cases of total case count)
       - confirmed no symptoms - total number of confirmed no symptoms cases (percentage of confirmed no symptoms cases of total case count)
       - confirmed unknown symptoms - total number of confirmed unknown symptoms cases (percentage of confirmed unknown symptoms cases of total case count)
       - not a case - total number of not a case cases (percentage of not a case cases or total case count)
    - Disease variants - total number of cases with disease variants
       - list of disease variants present in respective Sormas instance - total number of cases with respective variant (percentage of disease variants of total case count)

- Add the total number of contacts contained in the event and break these numbers further down
    - Unconfirmed contacts - total number of unconfirmed contacts (percentage of unconfirmed contacts of total contact count)
    - Confirmed contacts - total number of confirmed contacts (percentage of confirmed contacts of total contact count)
    - Converted to case - total number of converted contacts (percentage of converted contacts of total contact count)
    - Contact categories - total number of contacts 
        - High risk - total number of high risk contacts (percentage of high risk contacts of total contact count)
        - High risk med - total number of high risk med contacts (percentage of high risk med contacts of total contact count)
        - Medium risk med - total number of medium risk med contacts (percentage of medium risk med contacts of total contact count)
        - Low risk - total number of low risk contacts (percentage of low risk contacts or total contact count)

Due to displayed values being flexible because of either translations or customized enum values in e.g. disease variants, this info box should behave so it accommodates differently needed sizes.

Example does not necessarily include all values (see description)
<img width="664" alt="ParticipantSummary" src="https://user-images.githubusercontent.com/70317594/120150387-4a1fb780-c1eb-11eb-839b-515837ed1317.PNG">

### Possible Alternatives

If showing the summary has large performance implications, this could be initiated by clicking a button
<img width="728" alt="ParticipantSummaryAlternative" src="https://user-images.githubusercontent.com/70317594/120150431-586dd380-c1eb-11eb-8ec8-e256a3cb9c20.PNG">

**Comment by kwa20** [comment]
@JaquM @carolinverset @Marko-ilmkreis @bernardsilenou @ftavin Feedback is highly appreciated.

@Candice-Louw Maybe you would be able to comment on how the info box needs to behave in order to accommodate discrepancies in text length due to certain values like disease variants which are deposited for each instance individually and therefore flexible in length.

**Comment by bernardsilenou** [comment]
@kwa20 This is a valid need to have. However, a sever limitation I see with this design is that it would definitely affect performance. 
I would suggest we move it to the sormas-stats  OR keep it in sormas but move it to any of the statistic and reporting sections. 
Issue #5525 that is about computing the variant only, we think it would affect performance and would not compute it automatically but users have to trigger it.

Another suggestion is to keep it as you suggested and do not compute the indicators automatically. Users may missus and still cause performance issues. 

A good practice is to clearly separate `data collection` from  `data analysis and reporting`. Just as the dashboard for cases, we can use these proposed indicators as a basis and design an event dashboard + options to visualize the dashboard based on single event or many events

**Comment by kwa20** [comment]
@bernardsilenou If these things are mere counts, would this really have performance implications? Don't we already do this with e.g. columns in directories or elsewhere?

**Comment by bernardsilenou** [comment]
@kwa20 
Each computation affect performance, it all depends on what we want to sacrifice.
If you have thousands of users and each of them open an event, then the app needs to compute those indicators and load them on the form, this may slow navigation from one tab to another. If the box is not calculated automatically, then it is not a big problem I suppose.

I think we need to hear from @MateStrysewske  and @ftavin also to know the best option.

**Comment by Marko-ilmkreis** [comment]
@kwa20 @bernardsilenou I also agree to have that statistics view in each event. What's about to have it not shown when opening the event. Maybe there is only a button "show statistic for these event". Only when the user click this button sormas computes the figures.

**Comment by bernardsilenou** [comment]
@Marko-ilmkreis @kwa20  If the statistics is computed on users request, then its fine, even if it takes additional few seconds, they could not be discouraged since they triggered it.

**Comment by kwa20** [comment]
I've updated the initial description to include an alternative

**Comment by carolinverset** [comment]
@bernardsilenou @kwa20 @emond-fuseau
Sorry for my late answer... 
It's a really good idea and would bring real added value to users. But indeed, the risks of impact on performance must be controlled. This button to trigger calculations is a good idea. To be tested over the duration of treatment after activation. if there is a time out problem, the teams will be instructed not to activate it, which could lead to frustration. Perhaps make this button configurable so as to give the possibility of not showing this button in the application if the impact is too negative on performance.

**Comment by kwa20** [comment]
@carolinverset good idea, I've updated the initial description to include a feature configuration.
